js/resources_test.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import { test, testPerm, assert, assertEqual } from "./test_util.ts";
+import * as deno from "deno";
+
+test(function stdioResources() {
+  const res = deno.resources();
+
+  assertEqual(res[0].rid, 0);
+  assertEqual(res[0].repr, "stdin");
+
+  assertEqual(res[1].rid, 1);
+  assertEqual(res[1].repr, "stdout");
+
+  assertEqual(res[2].rid, 2);
+  assertEqual(res[2].repr, "stderr");
+});
+
+testPerm({ net: true }, async function netResources() {
+  const addr = "127.0.0.1:4500";
+  const listener = deno.listen("tcp", addr);
+
+  listener.accept().then(async conn => {
```

Comment:
Use await

Reply:
I took it from `net_test.ts`, IMHO it simpler to use `.then` here, otherwise we need some helper to do the `dial`.

Reply:
I see. Ok.